Add Python parser tests

Joshua2000B opened the pull request:
Adds the Python tests from the grammars repo. The tests are updated to the modern standard, and the majority of them currently pass. The rest (~23%) fail due to bugs/lack of support.

Joshua2000B committed:
Add Python parser tests

They are currently half-way done, they need verification to ensure the expected XML is correct

Joshua2000B committed:
Remaining Python testsuite files

Kyle776R reviewed (changes requested):
Overall, the test contents look fine; most comments are on tests to add. Here is a list of what should change before the tests are merged:

# Tests
- Add a few nested statement examples to `block_py.py.xml` (e.g., nested ifs or functions in a class) to ensure the block logic is working correctly.
	- There should be a test file for whitespace and comments located in miscellaneous places (e.g., whitespace + a comment after the start of a block, blank lines with only whitespace underneath functions, etc.) to ensure the block logic is working correctly.
	- There should be a test file with multiple blank lines located at the beginning, middle, and end of a file (to ensure the block logic is working correctly). This could either be its own test or combined with the test directly above this one.
- Add triple-quoted string examples to `literal_py.py.xml`.
- We need a test file for expressions that span more than one line (e.g., arrays, dictionaries, sets, etc.). These could go in their respective test files, but it might be easier to find them if they are in their own test file.

# Miscellaneous
- Add `revision` tags (1.0.0) to each individual unit. `call.preproc.c.xml` has an example of this.
- Remove the `<!-- 1 -->`, etc. comments before each individual unit. Some files have these, and some don't. Remove them for consistency with existing test files.

Joshua2000B committed:
Add some more tests.

There are still a few more left

Joshua2000B committed:
Final baseline changes to Python testsuite

Joshua2000B committed:
Add revision attributes to all Python tests

Also add some more niche test cases, like nested functions and one-line statements

Joshua2000B commented:
Tests have been greatly expanded. I added longer examples from the dummy files I was provided (not all of them - some were invalid Python). I also added tests for nested functions within the block test, multi-line calls in the call test, as well as a single line if, while, and function in their respective tests.
